app/controllers/preupgrade_reports_controller.rb

```diff
@@ -2,6 +2,14 @@
 
 class PreupgradeReportsController < ::Api::V2::BaseController
   def index
-    @preupgrade_reports = resource_scope.search_for(*search_options)
+    @preupgrade_reports = PreupgradeReport.search_for(*search_options)
```

Comment:
This should check `view_host` permission as well

Reply:
But why? We just list preupgrade records, no hosts. For `Api:V2::JobInvocationsController index` permission `:view_host` is also not checked.

Reply:
> We just list preupgrade records, no hosts.

Each preupgrade report is directly tied to a single host. If the users can't see the host, they shouldn't be able to see its preupgrade report. So if someone calls the index, they should get reports only for the hosts they can see.

> For Api:V2::JobInvocationsController index permission :view_host is also not checked

Because a job has many hosts and the index endpoint doesn't give any host-specific information. 

>  Same for the index action in Api::V2::PreupgradeReportsController

I'd say it should check it as well